src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -265,6 +265,109 @@ public static async ValueTask<FormulaValue> SortTable(EvalVisitor runner, Symbol
                 return SortValueType<BooleanValue, bool>(pairs, irContext, compareToResultModifier);
             }
         }
+        
+        public static async ValueTask<FormulaValue> SortByColumns(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
+        {
+            var table = (TableValue)args[0];
+            var tableType = (TableType)table.Type;
+
+            var records = table.Rows.ToList();
+            var errors = new List<ErrorValue>();
+            for (var argIndex = 1; argIndex < args.Length; argIndex += 2)
+            {
+                var fieldName = ((StringValue)args[argIndex]).Value;
+                if (tableType.MaybeGetFieldType(fieldName) is not NumberType and not StringType
+                    and not BooleanType and not DateTimeType and not DateType)
+                {
+                    // column not found error
+                    errors.Add(CommonErrors.RuntimeTypeMismatch(irContext));
+                    break;
+                }
+            }
+
+            if (errors.Count != 0)
+            {
+                return ErrorValue.Combine(irContext, errors);
+            }
+
+            foreach (var record in records)
+            {
+                if (record.IsError)
+                {
+                    errors.Add((ErrorValue)record.ToFormulaValue());
```

Comment:
Is this the correct error behavior? @CarlosFigueiraMSFT

Reply:
Yes - this is correct. When sorting, if we find any errors then the result will be an error. This implementation is returning all errors as it should.